src/client/compound/compound-bridge-data.ts

```diff
@@ -0,0 +1,150 @@
+import { EthAddress } from "@aztec/barretenberg/address";
+import { EthereumProvider } from "@aztec/barretenberg/blockchain";
+import { Web3Provider } from "@ethersproject/providers";
+import { BigNumber } from "ethers";
+import { ICERC20__factory, IComptroller__factory, IERC20__factory } from "../../../typechain-types";
+import { createWeb3Provider } from "../aztec/provider";
+import {
+  AssetValue,
+  AuxDataConfig,
+  AztecAsset,
+  AztecAssetType,
+  BridgeDataFieldGetters,
+  SolidityType,
+} from "../bridge-data";
+
+export class CompoundBridgeData implements BridgeDataFieldGetters {
+  expScale = 10n ** 18n;
+  allMarkets?: EthAddress[];
+
+  private constructor(private ethersProvider: Web3Provider) {}
+
+  static create(provider: EthereumProvider) {
+    return new CompoundBridgeData(createWeb3Provider(provider));
+  }
+
+  auxDataConfig: AuxDataConfig[] = [
+    {
+      start: 0,
+      length: 64,
+      solidityType: SolidityType.uint64,
+      description: "AuxData determine whether mint (0) or redeem flow (1) is executed",
+    },
+  ];
+
+  async getAuxData(
```

Comment:
Should this also take into account that assets need to be supported? e.g., if not supported throw?

Reply:
Good idea. Will do that 👍